tests/lib/e2e/users/users.e2e-spec.ts

```diff
@@ -0,0 +1,220 @@
+      })
+    })
+
+    describe('User creation v2', () => {
+      it('should create a user with email and password', async () => {
+        // Act: Create a user
+        const response = await testServer
+          .post('/v2/users')
+          .send({ email: 'test_v2' + testUser.email, password: testUser.hashpass })
+
+        // Assert
+        expect(response.status).toBe(200);
+        // expect(response.status).toBe(201);
+        const users = await engine.storage.models.User.find({ _id: response.body.id })
+        expect(users).toHaveLength(1)
+
+        expect(users[0].toObject().activated).toBe(true)
+      })
+    })
+
+    describe('User deletion v2', () => {
+
+      it('should be able to request a user deactivation', async () => {
+
+        // Arrange: Create User
+        const testEmail = 'request_deactivation_v2' + testUser.email
+        const { body: user } = await testServer
+          .post('/v2/users')
+          .send({ email: testEmail, password: testUser.hashpass, })
+          // .expect(201)
+          .expect(200);
+
+
+        // Act: Request Deactivation
+        const response = await testServer
+          .delete(`/v2/users/request-deactivate?deactivator=test-deactivator-token-request-deactivation-v2&redirect=/`)
+          .set('Authorization', getAuth({ email: testEmail, hashpass: testUser.hashpass }))
+
+        // Assert
+        expect(response.status).toBe(200);
+
+        const dbUser = await engine.storage.models.User.findOne({ _id: user.id })
+        expect(dbUser).not.toBeNull()
+
+        const token = dbUser.toObject().deactivator
+        expect(token).toBe('test-deactivator-token-request-deactivation-v2')
+
+        expect(sendGridMail.send).toHaveBeenCalled()
+
+
+      })
+
+      it('should be able to confirm a user deactivation', async () => {
+
+        // Arrange: Create User
+        const testEmail = 'confirm_deactivation_v2' + testUser.email
+        const { body: user } = await testServer
+          .post('/v2/users')
+          .send({ email: testEmail, password: testUser.hashpass, })
+          // .expect(201)
+          .expect(200);
+
+
+        // Arrange: Request Deactivation
+        const token = 'test-deactivator-token-confirm-deactivation-v2'
+        await testServer
+          .delete(`/v2/users/request-deactivate?deactivator=${token}&redirect=/`)
+          .set('Authorization', getAuth({ email: testEmail, hashpass: testUser.hashpass }))
+          .expect(200)
+
+
+        // Act: Confirm Deactivation
+        const response = await testServer.delete(`/v2/users/confirm-deactivate/${token}`)
+
+        // Assert
+        expect(response.status).toBe(200);
+        const dbUserAfterDeactivation = await engine.storage.models.User.findOne({ _id: user.id })
+        expect(dbUserAfterDeactivation).toBeNull()
+
+      })
+
+    })
+
+    describe('User update v2', () => {
+
+      it('should be able to update a user email via gateway', async () => {
```

Comment:
This PR still misses the product-oriented test legends, let's put an example here: 

```
describe('User updating its email', () => {
   it('When the user wants to change its email to a new free-to-use one, it works', () => {
     // some test...
   });
   
   it('When the user wants to change its email, but it already in use, it fails', () => {
    // some test
   })
   
   // and so on...
})
```

This way of writing the legends makes tests more readable and enforces it to be descriptive but not too technical if you go too much into detail, you will describe how the tested code does things, and if those things change, then you should change all the legends to be up-to-date. 

Adopting this technique also makes the test suite easy to understand and troubleshoot for other devs that may work in the future here, without knowing about the underlying implementation, but being able to spot what is failing, easily.